clients/osv/src/test/kotlin/ModelTest.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2022 EPAM Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.clients.osv
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.inspectors.forAll
+import io.kotest.matchers.shouldBe
+
+import java.io.File
+
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+
+class ModelTest : StringSpec({
+    "Deserializing and serializing any vulnerability is idempotent for all official examples" {
+        getVulnerabilityExamplesJson().forAll { vulnerabilityJson ->
+            val vulnerability = JSON.decodeFromString<Vulnerability>(vulnerabilityJson)
+
+            val serializedVulnerabilityJson = JSON.encodeToString(vulnerability)
+
+            normalizeJson(serializedVulnerabilityJson) shouldBe normalizeJson(vulnerabilityJson)
+        }
+    }
+})
+
+private val JSON = Json {}
+
+private fun getVulnerabilityExamplesJson(): List<String> =
+    (1..7).map { i -> File("src/test/assets/vulnerability/examples/$i.json").readText() }
```

Comment:
In order to not having to maintain this, how about just listing and iterating over all JSON files in the "src/test/assets/vulnerability/examples" directory?

Reply:
I had originally implemented what you propose, but then changed to this approach.
It has a little bit duplicate book keeping and ensures that tests don't get disabled by accident unnoticed.
MINOR: Furthermore it was easier to hard code a fixed asset (1..1) for development.

Can we keep it?

Reply:
> It has a little bit duplicate book keeping which ensures that tests don't get disabled by accident unnoticed.

Well, as you saw during the review using a number range does *not* prevent tests from getting disabled by accident, so I'm not convinced by that argument.

> Can we keep it?

TBH I just don't like it, and it's not in line with how we do it e.g. in [ExamplesFunTest](https://github.com/oss-review-toolkit/ort/blob/1ca71eb3c2ab8b8379eb9f387b13d0b66e276b83/cli/src/funTest/kotlin/ExamplesFunTest.kt#L66-L70), but I won't block it, esp. as you already have @mnonnenmacher's approval.